### DISCUSSION TITLE
Enabling logging for the phas4 and wss4j

### DISCUSSION THREAD

**Discussion by aks2687** [phax/phase4#108] [Q&A]
Hi, 
We are using the phase4 in our test project and added it as maven dependency. Execution is fine, but we are not able to enable the logs. While debugging found that it is returning NOPLogger for the phase4 and wss4j classes. We have just added it as a maven dependency in POM of our trial project. Is there anything we missed to enable the logging. 
Any help on this would be greate. 

Regards,
Akshay

**Comment by phax** [comment]
Hi Akshay,
Both wss4j and phase4 both use SLF4J for logging, which is a logging facade without actual logging functionality.
So what you need is to add e.g. Log4J2 or Logback and its respective SLF4J binding - see e.g. https://logging.apache.org/log4j/2.x/log4j-slf4j-impl/
Depending on your choice there, you can define the logging levels in the respective configuration file of the log framework you chose.
hth
